code/__HELPERS/unsorted.dm

```diff
@@ -1602,3 +1602,12 @@ config_setting should be one of the following:
 		if(-INFINITY to 0, 11 to INFINITY)
 			CRASH("Can't turn invalid directions!")
 	return turn(input_dir, 180)
+
+
+// This is used to cause less lag when you need to do excessive amount of process in a loop. (i.e. lubricate 10,000 tiles)
+/proc/lag_proof_sleep(maximum=200, sleep_time=1)
+	var/static/count = 0
+	count++
+	if(count >= maximum)
+		count=0
+		sleep(sleep_time)
```

Comment:
Whats wrong with just regular check_tick?

Reply:
It is a thing that should be actually implemented in cluwnificate tiles spell, but I got some minor derangement to put this. like the comment says, the game causes a lag when it has excessive amount of process regardless of its tick. - i.e. explosion of SM.
I had to put this into the cluwnificate tiles spell PR because it lubrciates some excessive amount of tiles that causes some lag that is similar behavior of crazy amount of monkies with hysteria virus. This is a concept of a solution that I found, but I think I should remove it from this PR for now then make something better.
at least I suspect it won't cause a bad lag for changing a lot of people's intents though... so removing it then making an another PR for this would be better, I guess.